src/version.cr

```diff
@@ -1,7 +1,7 @@
 module Shards
   VERSION    = {{ read_file("#{__DIR__}/../VERSION").chomp }}
   BUILD_SHA1 = {{ `git log --format=%h -n 1 2>/dev/null || echo ""`.stringify.chomp }}
-  BUILD_DATE = {{ `date -u +'%Y-%m-%d'`.stringify.chomp }}
+  BUILD_DATE = {{ `date -u --date="@${SOURCE_DATE_EPOCH:-$(date +%s)}" +'%Y-%m-%d'`.stringify.chomp }}
```

Comment:
We must not rely on `--date` which is not available on all platforms. This is a better implementation:

https://github.com/crystal-lang/crystal/blob/42a64909fae022863e68651a3d35f3c45884efc8/src/compiler/crystal/config.cr#L35-L36

Reply:
The updated code doesn't work because the result is not assigned to the constant. 

```cr
  BUILD_DATE = Time.unix({{ (env("SOURCE_DATE_EPOCH") || `date +%s`).to_i }})).to_s("%Y-%m-%d")
```

Reply:
Thanks updated the PR